### PR TITLE
Added login hook

### PR DESCRIPTION
**Pull request by razvantomegea**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import './styles/globals.css';
+import React from 'react';
 import { createRoot } from 'react-dom/client';
 import { App } from './App';
 
 const container = document.getElementById('root');
 const root = createRoot(container as HTMLElement);
-root.render(<App />);
+root.render((<App />) as any);
```
Comment (arhtudormorar) [inline src/index.tsx:8]: why as any?

File: src/localConstants/routes/routeNames.enums.ts [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+
 export enum RouteNamesEnum {
   home = '/',
   dashboard = '/dashboard',
   unlock = '/unlock',
   disclaimer = '/disclaimer',
-  send = '/send'
+  send = '/send',
+  sign = '/sign',
+  loginHook = `/hook/${HooksEnum.login}`
```
Comment (arhtudormorar) [inline src/localConstants/routes/routeNames.enums.ts:14]: add `HooksPageEnum.login`

File: src/pages/Unlock/Unlock.tsx [hunk 3]
```diff
@@ -12,22 +11,24 @@ import {
...
 
 export const Unlock = () => {
-  const navigate = useNavigate();
+  const onUnlockRedirect = useUnlockRedirect();
+
   const commonProps: CommonPropsType = {
-    callbackRoute: RouteNamesEnum.dashboard,
+    callbackRoute: routeNames.dashboard,
     nativeAuth,
     onLoginRedirect: () => {
```
Comment (arhtudormorar) [inline src/pages/Unlock/Unlock.tsx:30]: shorthand

File: src/pages/Unlock/hooks/useUnlockRedirect.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import { hookSelector } from 'redux/selectors';
+import { useOnLoginHookRedirect } from './useOnLoginHookRedirect';
+
+export const useUnlockRedirect = () => {
+  const isLoggedIn = useGetIsLoggedIn();
+  const onLoginHookRedirect = useOnLoginHookRedirect();
+  const navigate = useNavigate();
+  const { type: hook } = useSelector(hookSelector);
+  const { getRedirectPathname, pathname } = useRedirectPathname();
+  // const persistCrossWindowLogin = usePersistCrossWindowLogin();
```
Comment (arhtudormorar) [inline src/pages/Unlock/hooks/useUnlockRedirect.ts:15]: check comment

File: src/wrappers/AuthRedirectWrapper/AuthRedirectWrapper.tsx [hunk 5]
```diff
@@ -1,7 +1,9 @@
 import type { PropsWithChildren } from 'react';
+import { useSelector } from 'react-redux';
 import { Navigate } from 'react-router-dom';
 import { useGetIsLoggedIn } from 'hooks';
-import { RouteNamesEnum } from 'localConstants';
+import { routeNames } from 'routes';
+import { hookSelector } from '../../redux/selectors';
```
Comment (arhtudormorar) [inline src/wrappers/AuthRedirectWrapper/AuthRedirectWrapper.tsx:6]: absolute

File: src/components/Layout/Header/Header.tsx [hunk 6]
```diff
@@ -1,19 +1,51 @@
...
+import { Button } from 'components/Button';
 import { MxLink } from 'components/MxLink';
 import { environment } from 'config';
+import { logout } from 'helpers';
 import { useGetIsLoggedIn } from 'hooks';
-import { RouteNamesEnum } from 'localConstants';
 import { routeNames } from 'routes';
 import MultiversXLogo from '../../../assets/img/multiversx-logo.svg?react';
 
+const callbackUrl = `${window.location.origin}/unlock`;
```
Comment (arhtudormorar) [inline src/components/Layout/Header/Header.tsx:10]: undo

File: src/components/Status/Status.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import classNames from 'classnames';
+
+import { DataTestIdsEnum } from 'localConstants';
+import { getStatusIcon } from './helpers';
+import { StatusPropsType } from './status.types';
+
+export const Status = ({
```
Comment (arhtudormorar) [inline src/components/Status/Status.tsx:9]: StatusOfSomething ?

File: src/components/Utilities/components/PostMessageListener/helpers/buildTransactionsQueryString.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import { IPlainTransactionObject, Transaction } from '@multiversx/sdk-core/out';
+import { processBase64Fields } from '@multiversx/sdk-js-web-wallet-io/out/helpers';
+import { buildWalletQueryString } from './buildWalletQueryString';
+
+export const buildTransactionsQueryString = ({
```
Comment (arhtudormorar) [inline src/components/Utilities/components/PostMessageListener/helpers/buildTransactionsQueryString.ts:5]: is in wallet-io ?

File: src/components/Utilities/components/PostMessageListener/helpers/buildWalletQueryString.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import qs from 'qs';
+
+export const buildWalletQueryString = (options: { params?: any }): string => {
```
Comment (arhtudormorar) [inline src/components/Utilities/components/PostMessageListener/helpers/buildWalletQueryString.ts:3]: maybe is in wallet-io ?

File: src/helpers/hook/plainTransactionsToSearchString.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import { IPlainTransactionObject } from '@multiversx/sdk-core/out';
+import { processBase64Fields } from '@multiversx/sdk-js-web-wallet-io/out/helpers';
+import qs from 'qs';
+
+export const plainTransactionsToSearchString = (
```
Comment (arhtudormorar) [inline src/helpers/hook/plainTransactionsToSearchString.ts:5]: in wallet-io ?

File: src/localConstants/dataTestIds.enum.ts [hunk 11]
```diff
@@ -0,0 +1,4 @@
+export enum DataTestIdsEnum {
+  statusSubtitle = 'statusSubtitle',
```
Comment (arhtudormorar) [inline src/localConstants/dataTestIds.enum.ts:2]: check across project for dataTestId and insert here

File: src/localConstants/routes/routeNames.enums.ts [hunk 12]
```diff
@@ -1,8 +1,19 @@
...
+}
+
 export enum RouteNamesEnum {
   home = '/',
   dashboard = '/dashboard',
   unlock = '/unlock',
   disclaimer = '/disclaimer',
   send = '/send',
-  logout = '/logout'
+  logout = '/logout',
```
Comment (arhtudormorar) [inline src/localConstants/routes/routeNames.enums.ts:13]: should already be in main

File: src/localConstants/dataTestIds.enum.ts (hunk 11 above)
Comment (razvantomegea) [reply src/localConstants/dataTestIds.enum.ts:2]: In another PR/task

**Pull request merged**